pinot-common/src/main/java/org/apache/pinot/common/config/provider/TableCache.java

```diff
@@ -216,11 +241,19 @@ public boolean registerTableConfigChangeListener(TableConfigChangeListener table
   }
 
   /**
-   * Returns the schema for the given table, or {@code null} if it does not exist.
+   * Returns the schema for the given logical or physical table, or {@code null} if it does not exist.
    */
   @Nullable
   @Override
   public Schema getSchema(String rawTableName) {
+    if (isLogicalTable(rawTableName)) {
+      return getLogicalTableSchema(rawTableName);
+    } else {
+      return getPhysicalTableSchema(rawTableName);
+    }
+  }
+
+  private Schema getPhysicalTableSchema(String rawTableName) {
```

Comment:
(minor)
```suggestion
  private Schema getSchema(String schemaName) {
```

Reply:
`getSchema` already exists and it calls table specific get schema.

```
  public Schema getSchema(String rawTableName) {
    if (isLogicalTable(rawTableName)) {
      return getLogicalTableSchema(rawTableName);
    } else {
      return getPhysicalTableSchema(rawTableName);
    }
  }
```